MoreLinq.Test/RandomSubsetTest.cs

```diff
@@ -113,29 +121,33 @@ public void TestRandomSubsetShorterThanSequence()
         /// results in an exception. Only thrown when the resulting random sequence is enumerated.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         public void TestRandomSubsetLongerThanSequence()
         {
-            const int count = 100;
-            const int subsetSize = count + 5;
-            var sequence = Enumerable.Range(1, count);
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                const int count = 100;
+                const int subsetSize = count + 5;
+                var sequence = Enumerable.Range(1, count);
 
-            sequence.RandomSubset(subsetSize).Count();
+                sequence.RandomSubset(subsetSize).Count();
+            });
         }
 
         /// <summary>
         /// Verify that attempting to fetch a random subset longer than the original sequence
         /// results in an exception. Only thrown when the resulting random sequence is enumerated.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         public void TestRandomSubsetLongerThanSequence2()
         {
-            const int count = 100;
-            const int subsetSize = count + 5;
-            var sequence = Enumerable.Range(1, count);
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                const int count = 100;
+                const int subsetSize = count + 5;
+                var sequence = Enumerable.Range(1, count);
 
-            sequence.RandomSubset(subsetSize, new Random(1234)).Count();
+                sequence.RandomSubset(subsetSize, new Random(1234)).Count();
```

Comment:
Only this line should be inside the lambda to Assert.Throws as that's what will cause and being tested for an exception.